tests/TestCase.php

```diff
@@ -57,9 +58,19 @@ public function createApplication()
      */
     protected function refreshTestDatabase()
     {
-        Artisan::call('migrate:fresh', [
-            '--database' => 'explorer',
-            '--path'     => 'tests/migrations',
-        ]);
+        if (! RefreshDatabaseState::$migrated) {
+            $this->artisan('migrate:fresh', $this->migrateFreshUsing());
+
+            $this->artisan('migrate:fresh', [
+                '--database' => 'explorer',
+                '--path'     => 'tests/migrations',
+            ]);
+
+            $this->app[Kernel::class]->setArtisan(null);
+
+            RefreshDatabaseState::$migrated = true;
+        }
+
+        $this->beginDatabaseTransaction();
```

Comment:
Cant call parent::refreshTestDatabase because is a trait, so I copied the original logic + the explorer migration stuff